nix/per-system-outputs.nix

```diff
@@ -0,0 +1,24 @@
+# This file is part of the IOGX template and is documented at the link below:
+# https://www.github.com/input-output-hk/iogx#35-nixper-system-outputsnix
+
+{ nix, projects, ... }:
+
+rec {
+
+  packages.generated-puc-scripts-output =
+    nix.plutus-apps.generated-puc-scripts-outputs projects.default;
```

Comment:
There is a parameters to the interface files named `nix`, which can be used to reference files in your `./nix` folder by name using attribute sets, instead of using the `import` keyword. See this for documentation https://github.com/input-output-hk/iogx#301-nix

Reply:
You can support multiple ghc compilers in your project. (See `./nix/cabal-project.nix:supportedCompilers`).

So the `projects` argument will contain an attribute for each compiler name.
However you can use the `default` project which will use the `./nix/cabal-project.nix:defaultCompiler` (which is set the first compiler in `supportedCompilers` by default)

See here for more docs https://github.com/input-output-hk/iogx#351-projects